app/services/katello/pulp3/repository.rb

```diff
@@ -115,6 +121,19 @@ def list_remotes(args)
         remotes_api.list(args).results
       end
 
+      def self.delete_orphan_remotes(smart_proxy)
+        repo_types = pulp3_enabled_repo_types(smart_proxy)
+        repo_types.each do |repo_type|
+          pulp3_repo = repo_type.pulp3_service_class.new(nil, smart_proxy)
+          remotes = ::Katello::Pulp3::Repository.list_remotes(pulp3_repo, {})
```

Comment:
Were you able to try to refactor this based on our discussion to be able to access api classes via a class method instead of an instance variable?  (so we don't have to create the nil-based repo service class?)

Reply:
That's a work in progress. :)

Reply:
Attempted one route of the refactor, and was dissatisfied with the outcome. Still working on this.

Reply:
@jlsherrill ok I think I found a pattern that is aesthetically more pleasing than previous attempts. Take a look at https://github.com/Katello/katello/pull/8377/commits/696d6e812ee16dd2c1518ff1ece5a9ef66e0594c and let me know what you think.